lua/nvim-tree/legacy.lua

```diff
@@ -0,0 +1,175 @@
+local utils = require'nvim-tree.utils'
+
+local M = {}
+
+-- TODO update git.io/JPhyt when adding a migration
+
+local migrations = {
+  nvim_tree_disable_netrw = function(o)
+    if o.disable_netrw == nil then
+      o.disable_netrw = vim.g.nvim_tree_disable_netrw ~= 0
+    end
+  end,
+
+  nvim_tree_hijack_netrw = function(o)
+    if o.hijack_netrw == nil then
+      o.hijack_netrw = vim.g.nvim_tree_hijack_netrw ~= 0
+    end
+  end,
+
+  nvim_tree_auto_open = function(o)
+    if o.open_on_setup == nil then
+      o.open_on_setup = vim.g.nvim_tree_auto_open ~= 0
+    end
+  end,
+
+  nvim_tree_auto_close = function(o)
+    if o.auto_close == nil then
+      o.auto_close = vim.g.nvim_tree_auto_close ~= 0
+    end
+  end,
+
+  nvim_tree_tab_open = function(o)
+    if o.open_on_tab == nil then
+      o.open_on_tab = vim.g.nvim_tree_tab_open ~= 0
+    end
+  end,
+
+  nvim_tree_update_cwd = function(o)
+    if o.update_cwd == nil then
+      o.update_cwd = vim.g.nvim_tree_update_cwd ~= 0
+    end
+  end,
+
+  nvim_tree_hijack_cursor = function(o)
+    if o.hijack_cursor == nil then
+      o.hijack_cursor = vim.g.nvim_tree_hijack_cursor ~= 0
+    end
+  end,
+
+  nvim_tree_system_open_command = function(o)
+    utils.table_create_missing(o, "system_open")
+    if o.system_open.cmd == nil then
+      o.system_open.cmd = vim.g.nvim_tree_system_open_command
+    end
+  end,
+
+  nvim_tree_system_open_command_args = function(o)
+    utils.table_create_missing(o, "system_open")
+    if o.system_open.args == nil then
+      o.system_open.args = vim.g.nvim_tree_system_open_command_args
+    end
+  end,
+
+  nvim_tree_follow = function(o)
+    utils.table_create_missing(o, "update_focused_file")
+    if o.update_focused_file.enable == nil then
+      o.update_focused_file.enable = vim.g.nvim_tree_follow ~= 0
+    end
+  end,
+
+  nvim_tree_follow_update_path = function(o)
+    utils.table_create_missing(o, "update_focused_file")
+    if o.update_focused_file.update_cwd == nil then
+      o.update_focused_file.update_cwd = vim.g.nvim_tree_follow_update_path ~= 0
+    end
+  end,
+
+  nvim_tree_lsp_diagnostics = function(_)
+    -- option no longer exists
```

Comment:
this is `diagnostics.enable` i think

Reply:
Thank you. Added `diagnostics.enable` and `diagnostics.show_on_dirs`.

Did not test actual diagnostics, just that the opts were successfully set/ignored.

TIL diagnostics and will set them up for :make